pkg/controller/machinepool/machinepool_controller.go

```diff
@@ -582,20 +582,36 @@ func (r *ReconcileMachinePool) syncMachineSets(
 					}
 				}
 
-				// Update if the labels on the remote machineset are different than the labels on the generated machineset.
-				// If the length of both labels is zero, then they match, even if one is a nil map and the other is an empty map.
-				if rl, l := rMS.Spec.Template.Spec.Labels, ms.Spec.Template.Spec.Labels; (len(rl) != 0 || len(l) != 0) && !reflect.DeepEqual(rl, l) {
```

Comment:
PS: reflect.DeepEqual was removed for both cases because it is not an effective comparison - particularly for already existing labels/taints on remote machines if they need to be preserved